pkg/portfwd/client.go

```diff
@@ -24,26 +25,10 @@ func HandleTCPConnection(ctx context.Context, client *guestagentclient.GuestAgen
 		return
 	}
 
-	g, _ := errgroup.WithContext(ctx)
-
 	rw := &GrpcClientRW{stream: stream, id: id, addr: guestAddr}
-	g.Go(func() error {
-		_, err := io.Copy(rw, conn)
-		if errors.Is(err, io.EOF) {
-			return nil
-		}
-		return err
-	})
-	g.Go(func() error {
-		_, err := io.Copy(conn, rw)
-		if errors.Is(err, io.EOF) {
-			return nil
-		}
-		return err
-	})
 
-	if err := g.Wait(); err != nil {
-		logrus.Debugf("error in tcp tunnel for id: %s error:%v", id, err)
```

Comment:
The previous code tried to log both errors, assuming that if one copy fail, the other will fail soon, but this may not be the case. This may never log any error if the first io.Copy() fails and the other is stuck on blocking socket.

A better way to handle this is to use the same code we have in vz/network_darwin.go, logging the first error immediately.

Reply:
Sure. I moved all the logging back into bicopy.